File: Python/153.Find Minimum in Rotated Sorted Array .py
#在排序的循环数组中找到最小的数,二分法
class Solution(object):
    def findMin(self, nums):
        l , r = 0 , len(nums)-1
        while l<=r:
            mid = (l+r)//2
            if nums[mid]<nums[l]:
                l = l+1
                r = r-1
            elif nums[mid]<=nums[r]:
                r = mid-1
            else:
                l = mid+1
        return nums[mid]

s = Solution()
print(s.findMin([5,6,7,1,2,3,4]))

File: C++/153.Find Minimum in Rotated Sorted Array.cpp
/*
一个右移过的有序数组，寻找最小元素。
*/

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0, right = nums.size()-1 , mid;
        while(left<right)
        {
        	mid = (left + right)/2;
        	if(nums[mid]>=nums[left] && nums[mid]>=nums[right])
        	{
        		left = mid+1;
        	}
        	else
        	{
        		right = mid;
        	}
        }
        return nums[left];
    }
};

int main()
{
	Solution s;
	vector<int> nums;
	nums.push_back(4);
	nums.push_back(5);
	nums.push_back(6);
	nums.push_back(7);
	nums.push_back(0);
	nums.push_back(1);
	nums.push_back(2);
	cout<<s.findMin(nums)<<endl;
}